### PR TITLE
fix: Fix bug ResolveStructureValueAsync failing with TypeMismatch

### LINKED ISSUE #552: Bug Report: ResolveStructureValueAsync fails with TypeMismatch when backend returns JSON object
#### Description

When using `GetObjectValueAsync` or `GetObjectDetailsAsync` to retrieve a flag with an object/structure value, the provider throws a type mismatch error because it attempts to construct `new Value(JsonElement)` which is not a supported constructor overload.

#### Steps to Reproduce

1. Set up an OFREP-compliant backend that returns an object value:
```json
{
  "key": "my-object-flag",
  "value": {
    "property1": "value1",
    "property2": 123,
    "property3": true
  },
  "reason": "STATIC",
  "variant": "default"
}
```

2. Call `GetObjectValueAsync` from the client:
```csharp
var defaultValue = new Value(Structure.Builder()
    .Set("property1", "default")
    .Build());

var result = await featureClient.GetObjectValueAsync("my-object-flag", defaultValue, context);
var structure = result.AsStructure; // Fails before reaching here
```

3. Observe type mismatch error indicating `JsonElement` cannot be used.

#### Root Cause Analysis

The issue occurs in the interaction between `OfrepProvider.ResolveStructureValueAsync` and `OfrepClient.ProcessOkResponseAsync<T>`.

**Note:** The provider already handles `JsonElement` conversion for the `metadata` field via `MetadataExtensions.ToPrimitiveTypes()`, but this same pattern was not applied to the `value` field in `ResolveStructureValueAsync`.

**In `OfrepProvider.cs` (line 131-132):**
```csharp
var response = await this._client.EvaluateFlag(flagKey, defaultValue.AsObject,
    context, cancellationToken).ConfigureAwait(false);
```

The method passes `defaultValue.AsObject` which means `T = object`.

**In `OfrepClient.cs` (line 236):**
```csharp
resolvedValue = rawResponse.Value.Deserialize<T>(JsonOptions) ?? defaultValue;
```

When `T = object`, `JsonElement.Deserialize<object>()` returns the `JsonElement` itself (this is standard System.Text.Json behavior - it doesn't know what concrete type to deserialize to).

**Back in `OfrepProvider.cs` (line 138):**
```csharp
response.Value != null ? new Value(response.Value) : new Value(String.Empty)
```

`response.Value` is now a `JsonElement`, but `Value` has no constructor that accepts `JsonElement`.

#### Expected Behavior

`GetObjectValueAsync` should return a `Value` containing a properly constructed `Structure` that can be accessed via `.AsStructure`.

#### Suggested Fix

Add `JsonElement` to `Structure`/`Value` conversion in `ResolveStructureValueAsync`:

```csharp
public override async Task<ResolutionDetails<Value>> ResolveStructureValueAsync(
    string flagKey, Value defaultValue,
    EvaluationContext? context = null,
    CancellationToken cancellationToken = default)
{
    if (flagKey == null)
    {
        throw new ArgumentNullException(nameof(flagKey));
    }

    var response = await this._client.EvaluateFlag(flagKey, defaultValue.AsObject,
        context, cancellationToken).ConfigureAwait(false);

    Value resultValue;
    if (response.Value is JsonElement jsonElement)
    {
        resultValue = ConvertJsonElementToValue(jsonElement);
    }
    else if (response.Value != null)
    {
        resultValue = new Value(response.Value);
    }
    else
    {
        resultValue = defaultValue;
    }

    return new ResolutionDetails<Value>(
        flagKey,
        resultValue,
        MapErrorType(response.ErrorCode ?? string.Empty),
        reason: response.Reason,
        variant: response.Variant,
        errorMessage: response.ErrorMessage,
        flagMetadata: response.Metadata != null
            ? new ImmutableMetadata(response.Metadata.ToPrimitiveTypes())
            : null);
}

private static Value ConvertJsonElementToValue(JsonElement element)
{
    return element.ValueKind switch
    {
        JsonValueKind.Object => new Value(ConvertJsonElementToStructure(element)),
        JsonValueKind.Array => new Value(element.EnumerateArray()
            .Select(ConvertJsonElementToValue).ToList()),
        JsonValueKind.String => new Value(element.GetString()!),
        JsonValueKind.Number => element.TryGetInt64(out var longVal)
            ? new Value(longVal)
            : new Value(element.GetDouble()),
        JsonValueKind.True => new Value(true),
        JsonValueKind.False => new Value(false),
        _ => new Value()
    };
}

private static Structure ConvertJsonElementToStructure(JsonElement element)
{
    var builder = Structure.Builder();
    foreach (var property in element.EnumerateObject())
    {
        builder.Set(property.Name, ConvertJsonElementToValue(property.Value));
    }
    return builder.Build();
}
```

#### Environment

- OpenFeature.Providers.Ofrep: 0.1.4
- OpenFeature SDK: 2.11.1
- .NET: 9.0
- OS: Windows

### PR DESCRIPTION
**Pull request by askpt**
Signed-off-by: André Silva <2493377+askpt@users.noreply.github.com>

## This PR

This pull request improves the handling and conversion of JSON values returned from the OFREP backend, ensuring that JSON objects, arrays, numbers, and other types are correctly mapped to OpenFeature `Value` types. The changes resolve previous issues with type mismatches, particularly when dealing with JSON objects, and add comprehensive test coverage for these conversions.

Key changes include:

**Core JSON Conversion Logic:**

* Introduced a new `JsonConversionHelper` class to centralize the logic for extracting primitive values (string, int, double, bool, Dictionary, List, or null) from `JsonElement` instances, which is now used by both metadata and value conversion routines.
* Added `JsonElementExtensions` with a `ToValue` extension method, allowing seamless conversion from `JsonElement` to OpenFeature `Value`, including support for nested objects and arrays.

**Provider and Metadata Handling:**

* Updated `OfrepProvider` to request raw JSON (`JsonElement?`) from the backend and use the new conversion logic, preventing type mismatches and ensuring correct mapping to OpenFeature `Value` types.
* Refactored `MetadataExtensions` to use `JsonConversionHelper` for consistency and added logic to convert `null` values to empty strings for backwards compatibility in metadata.

**Testing Improvements:**

* Added a comprehensive test suite for `JsonElementExtensions`, covering conversion of all supported JSON types, including deeply nested structures, arrays, and edge cases.
* Enhanced provider tests to verify correct handling of JSON object values from the backend, including a test based on a real-world GitHub issue (#552). [[1]](diffhunk://#diff-fd8c7f0bbc49928ccf4dff5e35d5d902764ab141b5b092045c696eeea36a3767L210-R215) [[2]](diffhunk://#diff-fd8c7f0bbc49928ccf4dff5e35d5d902764ab141b5b092045c696eeea36a3767R226-R292)

**General Codebase Updates:**

* Updated relevant files to include necessary `System.Text.Json` imports for the new JSON handling logic. [[1]](diffhunk://#diff-f96424b489c48da1e498db0285dd1fa23042a1daa2ed70499a78213eee187050R1) [[2]](diffhunk://#diff-fd8c7f0bbc49928ccf4dff5e35d5d902764ab141b5b092045c696eeea36a3767R1)

These changes ensure robust and predictable handling of all JSON types returned from OFREP, improving compatibility and reliability for consumers of the provider.

### Related Issues

Fixes #552

### CODE REVIEW

File: src/OpenFeature.Providers.Ofrep/Extensions/MetadataExtensions.cs [hunk 1]
```diff
@@ -11,68 +11,47 @@ internal static class MetadataExtensions
...
-        var result = new List<object>();
-        foreach (var element in jsonElement.EnumerateArray())
-        {
-            result.Add(ExtractPrimitiveValue(element));
-        }
-        return result;
+            null => string.Empty,
+            Dictionary<string, object> dict => dict.ToDictionary(
+                kvp => kvp.Key,
+                kvp => ConvertNullsToEmptyString(kvp.Value) ?? string.Empty),
```
[the remark this answers is not in the document]
Comment (askpt) [reply src/OpenFeature.Providers.Ofrep/Extensions/MetadataExtensions.cs]: Fixed with: https://github.com/open-feature/dotnet-sdk-contrib/pull/556/commits/2f9e99a99afb204d2690bff9d74a132aa4a4f5e8
